src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -217,18 +217,17 @@ module.exports = function<T, P, I, TI, C>(
     const prevProps = current ? current.memoizedProps : null;
     let nextChildren = nextProps.children;
     const isDirectTextChild = shouldSetTextContent(nextProps);
+
     if (isDirectTextChild) {
       // We special case a direct text child of a host node. This is a common
       // case. We won't handle it as a reified child. We will instead handle
       // this in the host environment that also have access to this prop. That
       // avoids allocating another HostText fiber and traversing it.
       nextChildren = null;
-    } else if (prevProps && (
-      shouldSetTextContent(prevProps) ||
-      prevProps.children === null ||
```

Comment:
Is [this](https://github.com/bvaughn/react/blob/ae752dd5504c6d27a8e5cffeb2281705b2924623/src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js#L429-L443) what you had in mind, @acdlite?